src/abstractdataframe/selection.jl

```diff
@@ -86,22 +93,16 @@ function select_transform!(nc::Union{Pair{Int, Pair{ColRename, Symbol}},
     end
     if nc isa Pair{Int, Pair{ColRename, Symbol}}
         newdf[!, newname] = copycols ? df[:, col_idx] : df[!, col_idx]
-    elseif nc isa Pair{Int, <:Pair{<:Row, Symbol}}
-        newdf[!, newname] = (first(transform_spec).fun).(df[!, col_idx])
     elseif nc isa Pair{Int, <:Pair{<:Base.Callable, Symbol}}
         res = first(transform_spec)(df[!, col_idx])
         newdf[!, newname] = res isa AbstractVector ? res : [res]
-    elseif nc isa Pair{<:AbstractVector{Int}, <:Pair{<:Row, Symbol}}
-        if length(col_idx) == 0
+    elseif nc isa Pair{<:AbstractVector{Int}, <:Pair{<:Base.Callable, Symbol}}
+        if length(col_idx) == 0 && first(transform_spec) isa ByRow
```

Comment:
I was thinking this special case would be moved to ` function (f::ByRow)(....)`. Is that possible?

Reply:
It is not possible, as we would pass no columns to a function thus it would not have any way to know how many columns it has.

But now that I think about it actually it makes more sense to disallow passing no columns. I made a change that errors if user wants to select 0-columns and pass them to the function (then this part of code is simplified as we do not need this check).

We have the same problem with `combine`/`by`. I have opened https://github.com/JuliaData/DataFrames.jl/issues/2118 to decide on it.

Reply:
I guess one could want to do something like `select!(df, () => ByRow(rand) => :y)`?

I agree it's not a big deal, but it wouldn't be absurd to add this at some point. Unfortunately that would require special-casing `ByRow` here again, but that could be OK (maybe just passing it `axes(df, 1)` a bit like you did before).

Reply:
OK - for non-`ByRow` operations we can keep allowing 0-columns.

For `ByRow` related transforms we have a choice:
* allow passing 0-columns (but then I have to special case it as I did before)
* or disallow passing 0-columns with `ByRow`

I would be OK with both. Which one would you prefer?

Reply:
As you like. We can keep it that way for now and extend it later.

Reply:
OK. Then I will allow 0-columns for whole-column operations and disallow for `ByRow`.